ams/optimizer/src/main/java/com/netease/arctic/optimizer/util/K8sUtil.java

```diff
@@ -0,0 +1,162 @@
+        } else if (phase.equals("Running")) {
+          logger.info("Pod {} is running.", podName);
+          break;
+        }
+      }
+
+      // Check timeout
+      long elapsedTime = System.currentTimeMillis() - waitPodStartTime;
+      if (TimeUnit.MILLISECONDS.toSeconds(elapsedTime) > waitPodTimeout) {
+        throw new RuntimeException("Timeout reached. Job Pod create failed.");
+      }
+
+      // Polling interval
+      try {
+        TimeUnit.SECONDS.sleep(2);
+      } catch (InterruptedException e) {
+        throw new RuntimeException(e);
+      }
+    }
+    return podName;
+  }
+
+  public static void waitForDeleteJob(String namespace, String jobName, KubernetesClient client, Logger logger) {
+    long timeout = 300; // Timeout in seconds
+    long startTime = System.currentTimeMillis();
+    waitForDeleteJob(namespace, jobName, client, timeout, startTime, logger);
+  }
+
+  private static void waitForDeleteJob(String namespace, String jobName, KubernetesClient client, long timeout, long startTime, Logger logger) {
+
+    while (true) {
+      Job deletedJob = client.batch().v1().jobs()
+          .inNamespace(namespace)
+          .withName(jobName)
+          .get();
+
+      List<Pod> podList = client.pods()
+          .inNamespace(namespace)
+          .withLabel("job-name", jobName)
+          .list().getItems();
+
+      if (deletedJob == null && podList.isEmpty()) {
+        logger.info("Job {} has been deleted.", jobName);
+        return;
+      }
+
+      // Check timeout
+      long elapsedTime = System.currentTimeMillis() - startTime;
+      if (TimeUnit.MILLISECONDS.toSeconds(elapsedTime) > timeout) {
+        throw new RuntimeException("Timeout reached. Job deletion failed.");
+      }
+
+      // Polling interval
+      try {
+        TimeUnit.SECONDS.sleep(2);
+      } catch (InterruptedException e) {
+        throw new RuntimeException(e);
+      }
+    }
+  }
+
+  private static void submitJob(String namespace, String name, KubernetesClient client, String image, String cmd,String memory ,String cpu) {
+
+    Container container = new ContainerBuilder()
+        .withName("init")
+        .withImage(image)
+        .withCommand("sh", "-c", cmd)
+        .withResources(new ResourceRequirementsBuilder()
+            .withLimits(ImmutableMap.of("memory", new Quantity(memory), "cpu", new Quantity(cpu)))
+            .withRequests(ImmutableMap.of("memory", new Quantity(memory), "cpu", new Quantity(cpu)))
+            .build())
+        .build();
+
+    Job job = new JobBuilder()
+        .withNewMetadata()
+        .withName(name)
+        .endMetadata()
+        .withNewSpec()
+        .withBackoffLimit(0)
+        .withNewTemplate()
+        .withNewSpec()
+        .withContainers(container)
+        .withHostNetwork(true)
+        .withRestartPolicy("Never")
+        .endSpec()
+        .endTemplate()
+        .endSpec()
+        .build();
```

Comment:
Is using Job a good choice? Should Amoro Optimizer be a persistent computing resource?
Is Deployment or StatefulSet a better choice?